aspnet/fundamentals/routing.rst

```diff
@@ -1,10 +1,273 @@
 
-|stub-icon| Routing
-===================
+Sections:
+	- `Routing Middleware`_
+	- `Configuring Routing`_
+	- `Template Routes`_
+	- `Link Generation`_
+
+`View sample files <https://github.com/aspnet/Docs/tree/1.0.0-rc1/aspnet/fundamentals/routing/sample>`_
+
+Routing Middleware
+------------------
+The routing :doc:`middleware` uses *routes* to map requests to an ``IRouteHandler``. The ``IRouteHandler`` returns a ``RequestDelegate``, or ``null``. If it returns a delegate, that delegate is invoked; otherwise, routing continues. If no route handler is found for a request, then the middleware calls *next* (and the next middleware in the request pipeline is invoked).
+
+.. TODO in RC2 Middleware later in the request pipeline can access route data using extensions methods on ``HttpContext``.
+
+To get started using routing in your app, add it to the **dependencies** in ``project.json``:
+
+.. literalinclude:: routing/sample/RoutingSample/project.json
+  :linenos:
+  :dedent: 2
+  :language: javascript
+  :lines: 6-12
+  :emphasize-lines: 4
+  
+Next, add it to ``ConfigureServices`` in ``Startup.cs``:
+
+.. literalinclude:: routing/sample/RoutingSample/Startup.cs
+  :linenos:
+  :dedent: 8
+  :language: c#
+  :lines: 16-19
+  :emphasize-lines: 3
+
+With this in place, you will have access to routing in your app. The next step is to configure it.
+
+Configuring Routing
+-------------------
+Routing is configured in the ``Configure`` method in your ``Startup`` class. Create an instance of `RouteBuilder <https://github.com/aspnet/Routing/blob/1.0.0-rc1/src/Microsoft.AspNet.Routing/RouteBuilder.cs>`_. You can optionally set the ``ServiceProvider`` and/or ``DefaultHandler`` properties, in order to make them available as you build routes.
+
+.. literalinclude:: routing/sample/RoutingSample/Startup.cs
+  :linenos:
+  :dedent: 8
+  :language: c#
+  :lines: 21-34
+  :emphasize-lines: 7-8,10-13
+  
+You can see on line 13 how ``ApplicationServices`` is used to access the ``IInlineConstraingResolver`` dependency; if this were done from a ``RouteBuilder`` extension method, access to the services would be done through the ``ServiceProvider`` property.
+
+Once you've finished adding routes to the ``RouteBuilder`` instance, call ``UseRouter`` and pass it the result of the ``RouteBuilder.Build`` method.
+
+.. tip:: If you are only configuring a single route, you can simply call ``app.UseRouter`` and pass in the ``IRouter`` instance you wish to use, bypassing the need to use a ``RouteBuilder``.
+
+The sample shown above uses the default empty web template, which includes the "Hello World" request delegate shown on lines 16-19. The route added on line 8 will only match requests of the form "hello/{name}" where `name` is constrained to be alphabetical. Requests that match this will be handled by a custom ``IRouter`` implementation, ``HelloRouter``.
```

Comment:
Cleaned up the sentence to not refer to line numbers (especially since it was referencing line numbers that weren't even shown).